src/amber/router/file.cr

```diff
@@ -0,0 +1,26 @@
+require "tempfile"
+
+module Amber::Router
+  struct File
+    getter file : Tempfile
+    getter filename : String?
+    getter headers : HTTP::Headers
+    getter creation_time : Time?
+    getter modification_time : Time?
+    getter read_time : Time?
+    getter size : UInt64?
+
+    def initialize(upload)
+      @filename = upload.filename
+      @file = Tempfile.new(filename)
```

Comment:
@robacarp on this line however it could make sense to just have `@file = Tempfile.new`.